contracts/btc-light-client/src/msg/contract.rs

```diff
@@ -59,8 +59,16 @@ impl InstantiateMsg {
             return Err(ContractError::ZeroCheckpointFinalizationTimeout);
         }
 
-        // TODO: validate headers, first work and first height? For example, the base header
-        // should be on the difficulty boundary.
+        if !crate::bitcoin::is_difficulty_change_boundary(
+            self.initial_header.height,
+            &self.network.chain_params(),
+        ) {
+            return Err(ContractError::NotOnDifficultyBoundary(
+                self.initial_header.height,
+            ));
+        }
+
+        // TODO: the height should be larger than a recent block?
```

Comment:
Hmm it's BSN's responsibility to pick a recent height. If the BSN picks a height that is very old then the block sync will have some issues. So maybe this TODO is not needed

Reply:
That’s fair. I was just thinking it might be nice to catch some obvious misconfigurations in the BSN setup, hence the TODO with a question mark :( We can remove it for simplicity.